test-data/unit/check-optional.test

```diff
@@ -105,9 +105,12 @@ f(None)
 [out]
 main: note: In function "f":
 
-[case testInferOptionalType]
+[case testInferOptionalTypeWithBinder]
 x = None
-x = 1
+if 0:
```

Comment:
Is there a significance to the always-false-ness? Or is this just a convenient condition? I'd much rather see some random bool function used as the condition.